nemo/src/rule_model/components/rule.rs

```diff
@@ -74,34 +74,27 @@ impl Rule {
 
     /// Set how an instantiated version of the rule should be displayed.
     pub fn set_display(mut self, display: Term) -> Self {
-        self.display = Some(display);
+        self.description = Some(display);
         self
     }
 
-    /// Return a string representation of the rule instantiated with the given [Substitution].
-    /// This will either return
-    ///     * The content of the display attribute for this rule
-    ///     * a canonical string representation of the rule (i.e. [Display] representation)
-    /// whichever is the first defined in this list.
-    pub fn display_instantiated(&self, substitution: &Substitution) -> String {
-        if let Some(mut display) = self.display.clone() {
+    /// Return the name of the rule, if it is given one.
+    pub fn name(&self) -> Option<String> {
+        self.name.clone()
+    }
+
+    /// Return a string representation of the rule's description with the given [Substitution].
+    ///
+    /// Returns `None` if `description` results not in a ground term after applying the substitution.
+    pub fn description_instantiated(&self, substitution: &Substitution) -> Option<String> {
+        if let Some(mut display) = self.description.clone() {
```

Comment:
The local variable should likely also be called `description` to avoid confusion.